src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,52 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List</p>
  */
 public class ArrayList<T> implements List<T> {
+    private int size;
+    private int defaultCapacyty = 10;
+    private Object[] arr;
+
+    public ArrayList(int currentCapacity) {
+        if (currentCapacity > 0) {
+            this.arr = new Object[currentCapacity];
+        } else {
+            throw new IllegalArgumentException("Wrong initial capacity: " + currentCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.arr = new Object[defaultCapacyty];
+    }
+
+    private void setCapacity() {
+        if (size >= arr.length) {
+            arr = Arrays.copyOf(arr, (int) (arr.length * 1.5));
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        setCapacity();
+        arr[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        setCapacity();
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        System.arraycopy(arr, index, arr, index + 1, size - index);
+        arr[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
optional:
see how this method is implemented in the source code

Reply:
saw